### DISCUSSION TITLE
Keep getting errors after database changed

### DISCUSSION THREAD

**Discussion by arn4** [firefly-iii/firefly-iii#6621] [Firefly III]
Hello!
I've recently moved my Firefly database from remote to local installation. I took some time, but finally FireFly is up and working! 
The only problem is that every minute I get an email with this kind of error:
```
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '12' for key 'oauth_clients.PRIMARY' (SQL: insert into `oauth_clients` (`user_id`, `name`, `secret`, `provider`, `redirect`, `personal_access_client`, `password_client`, `revoked`, `updated_at`, `created_at`) values (?, FireflyIII Personal Access Client, [CODE EDITED FOR NOT SHARING DATA], ?, http://localhost/, 1, 0, 0, 2022-11-08 16:51:04, 2022-11-08 16:51:04))
```
I've noticed that `Duplicate entry '12'` is increasing at each message, while the coded I've remove from the log it's different every time. 

The app that was using this OAuth stopped connecting to the server, so I've removed all the clients manually from the web interface, but still I'm getting this error. What should I do? Maybe emptying that database table works, but I'm fearing to break my installation.

**Comment by JC5** [comment]
You can safely empty that particular table, although I'm not sure what the cause could be.

**Comment by arn4** [comment]
After 2 hours of one mail per minute with the error, it stopped without doing anything... Tried to reboot the server, but no new errors. No idea what happened.

Do you think I should do something or if it's working then it's ok?

**Reply by JC5** [reply]
If it ain't broke, don't fix it ;)

**Comment by arn4** [comment]
Hello again! After some weeks of smooth working, I got another error from the database. It was the first night where an automatic transaction was created after database migration.

Here is the error:
`Firefly III v5.7.6 ran into an error: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '1' for key 'notes.PRIMARY' (SQL: insert into `notes` (`noteable_id`, `noteable_type`, `text`, `updated_at`, `created_at`) values (651, FireflyIII\Models\TransactionJournal, Created from recurring transaction "Salary" (#3), 2022-11-25 03:00:01, 2022-11-25 03:00:01)).`

In the end the transaction was created correctly, expect for the fact that when I click on it in FireFly I get 404 error (from the link, the transaction ID is 0). I don't know what to do because I'd like to edit it,but I can't. Moreover I'm worried what could happen after next automatic transaction will be created.

Any idea on how to fix and prevent this to happen again?

**Reply by JC5** [reply]
Firefly III doesn't assign ID's, it just sends the data to the database. I must admit I have no idea how to fix this.

**Reply by arn4** [reply]
I've investigated a little bit and found that the entry in `transaction_groups` was missing, while both `transaction_journals` and `transactions` were fine. I solved by manually deleting the entries in the latter two tables from MySQL console. Then I've just created the new transaction from FireFly and now everything is working again. Let's wait next month to see if fails again.

**Reply by JC5** [reply]
I'm curious to see what will happen!

**Comment by arn4** [comment]
Follow up from previous discussion. I think I've finally understood the cause of the problems: every time I try to create a transaction with notes I get an error of this kind:
```Error! Internal Firefly III Exception: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '8' for key 'notes.PRIMARY' (SQL: insert into `notes` (`noteable_id`, `noteable_type`, `text`, `updated_at`, `created_at`) values (704, FireflyIII\Models\TransactionJournal, Random text edited for privacy, 2022-12-19 22:48:43, 2022-12-19 22:48:43))```
This was also the problem with automatic transactions.

I really don't know what to do. For me it's not a big deal to lose all transaction notes up to now, but I'd like to be able to use notes again. Do you have any suggestion on what I can do? Some kind of database repair I can try?

**Reply by JC5** [reply]
This is MySQL, right? That's something you can Google, I think.
